connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/feel/jackson/FeelSupplierDeserializer.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.runtime.core.feel.jackson;
+
+import com.fasterxml.jackson.databind.BeanProperty;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import io.camunda.connector.runtime.core.feel.FeelEngineWrapper;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public class FeelSupplierDeserializer<OUT> extends AbstractFeelDeserializer<Supplier<OUT>> {
+
+  Class<OUT> outputType;
+
+  protected FeelSupplierDeserializer(Class<OUT> outputType, FeelEngineWrapper feelEngineWrapper) {
+    super(feelEngineWrapper);
+    this.outputType = outputType;
+  }
+
+  @Override
+  Supplier<OUT> doDeserialize(String expression) {
+    // evaluate eagerly to fail fast
+    OUT result = feelEngineWrapper.evaluate(expression, Map.of(), outputType);
+    return () -> result;
```

Comment:
Shouldnt we only evaluate expressions eagerly if the target type is neither Function nor Supplier?
I would therefore see three case for the different target types:
* Supplier -> eval lazy
* Function -> eval during application
* Everything else -> eval eagerly (this can be used to directly bind a FEEL expression to a List<String> for example.

Reply:
How would we implement a custom deserializer for the third case? As mentioned, I didn't find a way to contextually deserialize a String into generic type if that string is not guaranteed to be a FEEL expression (if it's just a string, we would need to somehow delegate to another deserializer which we don't know about).

Re. the eager evaluation for supplier, I think it makes sense to validate it and fail fast before running the connector, as it's just an expression without arguments

Reply:
@chillleader We would only eagerly evaluate if we tested whether the string is a FEEL expression (as we do already) and then evaluate and bind the result right away.

Reply:
I implemented the deserializer you mentioned (`FeelDeserializer`) but couldn't find a way to register it for any property due to Jackson's deserializer selection precedence. So I suggest that we manually enable this deserializer for properties where eager FEEL expression evaluation is required, like this: `@JsonDeserialize(using = FeelDeserializer.class)`.

I think this approach would be better than to wrap the values that need to parsed into some other wrapper type or annotating them with a custom annotation. And if the future we find a way to enable this resolution automatically, the old way with the standard `@JsonDeserialize` annotation will still work and will be easy to migrate from. WDYT?

Reply:
I think thats a good comprosmise for what we initially planned. Great effort and thanks for further investigating the topic.